rules/experimental/rbe/exec_properties.bzl

```diff
@@ -0,0 +1,163 @@
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""This file contains macros to create and manipulate dictionaries of properties to be used as execution properties for RBE.
+"""
+
+def _add(
+        dict,
+        var_name,
+        key,
+        value,
+        verifier_fcn = None,
+        transform_fcn = None):
+    """Add a key-value to a dict.
+
+    If value is None, don't add anything.
+    The dict will always be a string->string dict, but the value argument to this function may be of a different type.
+
+    Args:
+      dict: The dict to update.
+      var_name: Used for error messages.
+      key: The key in the dict.
+      value: The value provided by the caller. This may or may not be what ends up in the dict.
+      verifier_fcn: Verifies the validity of value. On error, it's the verifier's responsibility to call fail().
+      transform_fcn: Transform the value provided to this function with a value to put in the dict.
+          Note that a transform_fcn is not needed for casting a non string to string, this will be done anyway.
+    """
+    if value == None:
+        return
+    if verifier_fcn != None:
+        verifier_fcn(var_name, value)  # verifier_fcn will fail() if necessary
+    if transform_fcn != None:
+        value = transform_fcn(value)
+    dict[key] = str(value)
+
+def _verify_string(var_name, value):
+    if type(value) != "string":
+        fail("%s must be a string" % var_name)
+
+def _verify_bool(var_name, value):
+    if type(value) != "bool":
+        fail("%s must be a bool" % var_name)
+
+def _verify_os(var_name, value):
+    _verify_string(var_name, value)
+    valid_os_list = ["Linux", "Windows"]
+    if value not in valid_os_list:
+        fail("%s must be one of %s" % (var_name, valid_os_list))
+
+def _transform_network(value):
+    return "standard" if value else "off"
+
+def create_exec_properties_dict(**kwargs):
+    """Return a dict with exec_properties that are supported by RBE.
+
+    Args:
+      **kwargs: Arguments specifying what keys are populated in the returned dict.
+          Note that the name of the key in kwargs is not the same as the name of the key in the returned dict.
+          For more information about what each parameter is see https://cloud.google.com/remote-build-execution/docs/remote-execution-environment#remote_execution_properties.
+          If this link is broken for you, you may not to be whitelisted for RBE. See https://groups.google.com/forum/#!forum/rbe-alpha-customers.
+
+    Returns:
+      A dict that can be used as, for example, the exec_properties parameter of platform.
+    """
+    params = {
```

Comment:
you can probably declare this statically outside of the def. But that can be fine in an upcoming PR to unblock you.

Reply:
Done.